src/index.js

```diff
@@ -195,69 +171,48 @@ export default class HyperScreen extends React.Component {
-      .then((response) => {
-        if (!response.ok) {
-          throw Error(response.statusText);
-        }
-        return response.text();
-      })
-      .then((responseText) => {
-        if (typeof this.props.onParseBefore === 'function') {
-          this.props.onParseBefore(url);
-        }
-        let doc = this.parser.parseFromString(responseText);
-        let error = false;
-        const stylesheets = Stylesheets.createStylesheets(doc);
-        this.navigation.setRouteKey(url, routeKey);
-
-        // Make sure the XML has the required elements: <doc>, <screen>, <body>.
-        const docElement = getFirstTag(doc, 'doc');
-        if (!docElement) {
-          console.error(`No <doc> tag found in the response from ${url}.`);
-          doc = null;
-          error = true;
-        } else {
-          const screenElement = getFirstTag(docElement, 'screen');
-          if (!screenElement) {
-            console.error(`No <screen> tag found in the <doc> tag from ${url}.`);
-            doc = null;
-            error = true;
-          } else {
-            const bodyElement = getFirstTag(screenElement, 'body');
-            if (!bodyElement) {
-              console.error(`No <body> tag found in the <screen> tag from ${url}.`);
-              doc = null;
-              error = true;
-            }
-          }
-        }
+    try {
+      if (params.delay) {
+        await later(parseInt(params.delay, 10));
+      }
 
-        this.setState({
-          doc,
-          styles: stylesheets,
-          error,
-        });
-        if (typeof this.props.onParseAfter === 'function') {
-          this.props.onParseAfter(url);
-        }
-      })
-      .catch((reason) => {
-        this.setState({
-          error: true,
-        });
-        throw reason;
+      const url = this.state.url;
+      const doc = await this.parser.load(url);
+
+      // Make sure the XML has the required elements: <doc>, <screen>, <body>.
+      const docElement = getFirstTag(doc, 'doc');
+      if (!docElement) {
+        throw new Error(`No <doc> tag found in the response from ${url}.`);
+      }
+
+      const screenElement = getFirstTag(docElement, 'screen');
+      if (!screenElement) {
+        throw new Error(`No <screen> tag found in the <doc> tag from ${url}.`);
+      }
+
+      const bodyElement = getFirstTag(screenElement, 'body');
+      if (!bodyElement) {
+        throw new Error(`No <body> tag found in the <screen> tag from ${url}.`);
+      }
+
+      const stylesheets = Stylesheets.createStylesheets(doc);
+      this.navigation.setRouteKey(url, routeKey);
+      this.setState({
+        doc,
+        styles: stylesheets,
+        error: false,
       });
 
-    if (delay) {
-      later(parseInt(delay, 10)).then(fetchPromise);
-    } else {
-      fetchPromise();
+    } catch (err) {
+      this.setState({
+        doc: null,
+        styles: null,
+        error: true,
+      });
+      console.error(err.message);
```

Comment:
Personally, I consider it bad form for a package to write to the console on its own. If there's a logging abstraction, we should use that - but otherwise, I think we should either leave logging to the consumer, or throw an error they have to handle.

Also, is there a reason we can't include the actual error in the state update here? Maybe just set `error: err` instead of `true`?

Reply:
a couple thing with this regards:
- `console.error` in react native triggers the "redbox" alert on the device, only for debug builds - it's noop for prod builds.
- it expects a string in order to do proper rendering of the error - just passing the error would effectively trigger a redbox with the error constructor name only and no details